diff --color -rupN uboot-2019.04.a/board/ti/am335x/mux.c uboot-2019.04.b/board/ti/am335x/mux.c
--- uboot-2019.04.a/board/ti/am335x/mux.c	2020-12-14 11:27:14.482982725 +0100
+++ uboot-2019.04.b/board/ti/am335x/mux.c	2020-12-14 11:29:40.459391379 +0100
@@ -437,4 +437,8 @@ void enable_board_pin_mux(void)
 		/* Unknown board. We might still be able to boot. */
 		puts("Bad EEPROM or unknown board, cannot configure pinmux.");
 	}
+
+	// preitinger: immer pin muxing fuer nand und i2c0 damit nand und eeprom verfuegbar fuer update-funktionalitaet 
+	configure_module_pin_mux(nand_pin_mux);
+	enable_i2c0_pin_mux();
 }
