diff --color -rupN mtd-2.1.1/ubi-utils/ubiformat.c mtd-2.1.1.b/ubi-utils/ubiformat.c
--- mtd-2.1.1/ubi-utils/ubiformat.c	2019-05-13 08:27:22.000000000 +0200
+++ mtd-2.1.1.b/ubi-utils/ubiformat.c	2020-12-15 17:52:10.235654806 +0100
@@ -550,6 +550,7 @@ static int format(libmtd_t libmtd, const
 	struct ubi_vtbl_record *vtbl;
 	int eb1 = -1, eb2 = -1;
 	long long ec1 = -1, ec2 = -1;
+	int ret = -1;
 
 	write_size = UBI_EC_HDR_SIZE + mtd->subpage_size - 1;
 	write_size /= mtd->subpage_size;
@@ -643,8 +644,10 @@ static int format(libmtd_t libmtd, const
 	if (!args.quiet && !args.verbose)
 		printf("\n");
 
-	if (novtbl)
+	if (novtbl) {
+		ret = 0;
 		goto out_free;
+	}
 
 	if (eb1 == -1 || eb2 == -1) {
 		errmsg("no eraseblocks for volume table");
@@ -669,7 +672,7 @@ static int format(libmtd_t libmtd, const
 
 out_free:
 	free(hdr);
-	return -1;
+	return ret;
 }
 
 int main(int argc, char * const argv[])
